src/components/CDPDisplay/index.js

```diff
@@ -8,25 +8,42 @@ import useStore from 'hooks/useStore';
 import { getCdp } from 'reducers/cdps';
 import { trackCdpById } from 'reducers/multicall/cdps';
 import CDPViewPresentation from './Presentation';
+import Unavailable from '../Unavailable';
+import { Routes, ZERO_ADDRESS } from '../../utils/constants';
+import { useNavigation } from 'react-navi';
 
 function CDPView({ cdpId }) {
   cdpId = parseInt(cdpId, 10);
   const { maker, account, network } = useMaker();
   const { show: showSidebar } = useSidebar();
   const [{ cdps, feeds }, dispatch] = useStore();
   const [cdpOwner, setOwner] = useState();
+  const [cdpAvailable, setCdpAvailable] = useState(true);
+  const navigation = useNavigation();
 
   useEffect(() => {
     (async () => {
+      async function redirect(account) {
+        const { search } = (await navigation.getRoute()).url;
+        navigation.navigate({
+          pathname: `/${Routes.BORROW}/owner/${account.address}`,
+          search
+        });
+      }
       const proxyAddress = await maker
         .service('mcd:cdpManager')
         .getOwner(cdpId);
```

Comment:
once you know that getOwner returns 0x0, can you redirect / setCdpAvailable(false) at that point, rather than doing the proxy owner check?